megamek/src/megamek/common/MoveStep.java

```diff
@@ -2953,16 +2953,20 @@ && getEntity().getPosition().equals(prev)
                     break;
             }
             // Light
-            switch (game.getPlanetaryConditions().getLight()) {
-                case PlanetaryConditions.L_FULL_MOON:
-                    mp += 1;
-                    break;
-                case  PlanetaryConditions.L_MOONLESS:
-                    mp += 2;
-                    break;
-                case PlanetaryConditions.L_PITCH_BLACK:
-                    mp += 3;
-                    break;
+            if (!entity.isNightwalker()) {
+                switch (game.getPlanetaryConditions().getLight()) {
+                    case PlanetaryConditions.L_FULL_MOON:
+                        mp += 1;
+                        break;
+                    case PlanetaryConditions.L_MOONLESS:
+                        mp += 2;
+                        break;
+                    case PlanetaryConditions.L_PITCH_BLACK:
+                        mp += 3;
+                        break;
+                }
+            } else {
+                setRunProhibited(true);
             }
```

Comment:
Does this ability let you choose if it's used or not? If so, this is likely a problematic way to handle it.

Reply:
The ability is supposed to allow for full walk movement at night OR run at a -1 to the night move penalty. Both Nick and I have tried to work in both but it can't be done.

As for choice, no. Say you have a 5/8 mech, w/o the SPA you are running 4 hexes at 2mp per hex minimum under a full moon. With the SPA you can walk 5 hexes.

The choice making this was either walk full speed or run at -1 to the penalty. Disabling running seemed the better option of the two for the player

Reply:
I've been thinking about this today, and it's a huge negative for daytime use. Also, do we not split movement between the various types?

Reply:
Derp, first is a bug. I rewrote this from memory and thought I had it right.

The second I'm not sure what you mean.

Reply:
How is movement calculated, especially the various levels of movement. I've never looked at it, so I'm wondering how it works as I have ideas for potentially handling it better.

Reply:
Ah, ok. I know weather MP modifiers are handled in each units class while terrain, fog, and night are handled in MoveStep.

MoveStep processes each hex a unit travels over and builds a list of mp spent moving into each hex. The terrain cost is in either Terrain or Terrains, I don't remember. Regardless the function processMovementCost handles all of it. Not sure what handles the color changing when going from walk to run to sprint. I'm not as familiar with it as I used to be so it is hard to properly explain, though with Cross-Country being half done I will have to figure out how it works better if I am to makes tanks swim.

Also, if you check in Tank, you will find functions for getWalkMP, getRunMP, getSprintMP etc that defines each from the rules. If this has been any help I would be surprised. I can read and follow the flow of the code I'm just not very good at explaining it. Nick or Arlith would be of more help. I don't know if neoancient ever touched this section of the code or not, so maybe him as well?